docs/Changelog.md

```diff
@@ -18587,6 +18587,94 @@ This version of the operator has been available since version 13 of the default
 </dl>
 
 ## Version 14 of the default ONNX operator set
+### <a name="BatchNormalization-14"></a>**BatchNormalization-14**</a>
+
+  Carries out batch normalization as described in the paper
+  https://arxiv.org/abs/1502.03167. Depending on the mode it is being run,
+  There is three required inputs 'X', 'mean' and 'var', in addition to one
+  optional input 'training_mode'.
+  Note that 'mean' and 'var' are expected to be the estimated statistics in
+  inference mode (training_mode=False, default),
+  and the running statistics in training mode (traning_mode=True).
+  There are multiple cases for the number of outputs, which we list below:
+  
+  Output case #1: Y, mean, var, saved_mean, saved_var (training_mode=True)
+  Output case #2: Y (training_mode=False)
+  
+  The output and statistics are updated as follows when training_mode=True:
+  ```
+  saved_mean = ReducedMean(X, axis=all_except_channel_index)
+  saved_var =  ReducedVar(X, axis=all_except_channel_index)
+  
+  output_mean = mean * momentum + saved_mean * (1 - momentum)
+  output_var = var * momentum + saved_var * (1 - momentum)
+  
+  Y = (X - saved_mean) / sqrt(var + saved_epsilon) * scale + B
+  ```
+  
+  When training_mode=False:
+  ```
+  Y = (X - mean) / sqrt(var + epsilon) * scale + B
+  ```
+  
+  For previous (depreciated) non-spatial cases, implementors are suggested
```

Comment:
I don't understand this either. If the inputs are required to have some shape, can we explicitly document it here? Is it required to be 2D?

Reply:
I think this is a note about how to get the old (deprecated) behavior for the op. There was an attribute in older versions:
'Spatial': If true, compute the mean and variance across per activation. If false, compute the mean and variance across per feature over each mini-batch.
And I think this note specifies how to achieve this behavior.

Input X can have N dimensions.
I can clarify the lines above.

Reply:
I think input `X` is expected to have at least rank 2. In the most common case input will be NCHW, but as noted in the comment to replicate the older non-spatial behavior you can flatten it to `N x CHW`.